src/Runner/TestSuiteSorter.php

```diff
@@ -31,27 +32,76 @@
      */
     public const ORDER_REVERSED = 2;
 
+    /**
+     * @var int
+     */
+    public const ORDER_DEFECTS_FIRST = 3;
+
+    /**
+     * List of sorting weights for all test result codes. A higher number gives higher priority.
+     */
+    private const DEFECT_SORT_WEIGHT = [
+        ResultCacheListener::TEST_SKIPPED    => 7,
+        ResultCacheListener::TEST_ERROR      => 6,
+        ResultCacheListener::TEST_FAILURE    => 5,
+        ResultCacheListener::TEST_WARNING    => 4,
+        ResultCacheListener::TEST_INCOMPLETE => 3,
+        ResultCacheListener::TEST_RISKY      => 2,
+        ResultCacheListener::TEST_UNKNOWN    => 0
+    ];
+
+    /**
+     * @var array[string]int Associative array of (string => DEFECT_SORT_WEIGHT) elements
+     */
+    private $defectSortOrder;
+
+    /**
+     * @var TestResultCache
+     */
+    private $cache;
```

Comment:
still.
```php
new TestSuiteSorter()
```

and this property is null, not `TestResultCache`

use `?TestResultCache` or `null|TestResultCache`

Reply:
There might be a few more of those, I will check and change my code style for this.